ai_user/ai_user.py

```diff
@@ -116,47 +118,45 @@ async def on_message_edit(self, before: discord.Message, after: discord.Message)
 
     async def is_common_valid_reply(self, ctx: commands.Context) -> bool:
         """ Run some common checks to see if a message is valid for the bot to reply to """
-        if not ctx.guild:
+        if not ctx.guild or ctx.author.bot:
             return False
-
         if await self.bot.cog_disabled_in_guild(self, ctx.guild):
             return False
-
-        if ctx.author.bot:
+        if not await self.bot.ignored_channel_or_guild(ctx):
+            return False
+        if not await self.bot.allowed_by_whitelist_blacklist(ctx.author):
             return False
 
         if not openai.api_key:
-            await self.initalize_openai(ctx.message)
-
+            await self.initalize_openai(ctx)
         if not openai.api_key:
             return False
 
         if not self.cached_options.get(ctx.message.guild.id):
             await self.cache_guild_options(ctx)
 
-        if isinstance(ctx.channel, discord.Thread):
-            if ctx.channel.parent.id not in self.cached_options[ctx.guild.id].get("channels_whitelist"):
+        if not ctx.interaction:
+            if isinstance(ctx.channel, discord.Thread):
```

Comment:
```suggestion
        if ctx.interaction:
            return True
        if isinstance(ctx.channel, discord.Thread):
```
Does this work or is there a reason we want it this way? Want to avoid triple-nesting if statements

Reply:
Yeah good call